### PR TITLE
Convert execute plugins to tmt.utils.field() for options

### PR DESCRIPTION
**Pull request by happz**
The `upgrade` plugin needed a bit more work, there is a stuff inherited from the `internal` plugin, but in general, things should be simpler now.

### CODE REVIEW

File: tmt/steps/execute/upgrade.py [hunk 1]
```diff
@@ -28,19 +27,45 @@
...
-
-    _normalize_test = tmt.utils.NormalizeKeysMixin._normalize_string_list
-    _normalize_filter = tmt.utils.NormalizeKeysMixin._normalize_string_list
-    _normalize_exclude = tmt.utils.NormalizeKeysMixin._normalize_string_list
+    ref: Optional[str] = field(
+        default=cast(Optional[str], None),
+        option=('-r', '--ref'),
+        metavar='REVISION',
+        help='Branch, tag or commit specifying the git revision.')
+    filter: List[str] = field(
```
Comment (psss) [inline tmt/steps/execute/upgrade.py:48]: Seems that the `test` and `filter` options are swapped here?

File: tmt/steps/execute/upgrade.py [hunk 2]
```diff
@@ -112,29 +137,13 @@ class ExecuteUpgrade(ExecuteInternal):
     """
 
     # FIXME: ignore[assignment]: https://github.com/teemtee/tmt/issues/1540
```
Comment (psss) [inline tmt/steps/execute/upgrade.py:139]: Comment to be removed as well?

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for the cleanup! Looks good except for the options swap.

File: tmt/steps/execute/upgrade.py (hunk 1 above)
Comment (psss) [reply tmt/steps/execute/upgrade.py:48]: Hmmm, it's a pitty that tests did not catch this.

File: tmt/steps/execute/upgrade.py (hunk 2 above)
Comment (happz) [reply tmt/steps/execute/upgrade.py]: Fixed in 57fc488a.

File: tmt/steps/execute/upgrade.py (hunk 1 above)
Comment (happz) [reply tmt/steps/execute/upgrade.py]: Fixed in 57fc488a.

**Pull request merged**